### PR TITLE
Change routes for Experience Points Records

### PR DESCRIPTION
**Pull request by kxmbrian**
Fix for #914

### CODE REVIEW

File: app/controllers/course/experience_points_records_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 class Course::ExperiencePointsRecordsController < Course::ComponentController
-  load_resource :course_user, through: :course
+  before_action :load_course_user
```
Comment (lowjoel) [inline app/controllers/course/experience_points_records_controller.rb]: why change this? you've changed the meaning of the call.

File: app/views/course/experience_points_records/_experience_points_record.html.slim [hunk 2]
```diff
@@ -3,4 +3,6 @@
   td = format_inline_text(experience_points_record.reason)
   td = link_to_user(experience_points_record.creator)
   - if experience_points_record.manually_awarded? && can?(:destroy, experience_points_record)
-    td = delete_button([current_course, experience_points_record.course_user, experience_points_record])
+    td = delete_button(course_user_experience_points_record_path(current_course,
+                                                                 @course_user,
```
Comment (lowjoel) [inline app/views/course/experience_points_records/_experience_points_record.html.slim]: `current_course_user`

File: config/routes.rb [hunk 3]
```diff
@@ -217,6 +217,7 @@
       resources :users, only: [:index, :show, :update, :destroy] do
         get 'invite' => 'user_invitations#new', on: :collection
         post 'invite' => 'user_invitations#create', on: :collection
+        resources :experience_points_records, only: [:index, :destroy]
```
Comment (lowjoel) [inline config/routes.rb]: place above invite.

File: app/controllers/course/experience_points_records_controller.rb [hunk 4]
```diff
@@ -19,16 +20,20 @@ def destroy # :nodoc:
 
   private
 
+  # Set +course_user_id+ in +params+ so that cancancan can automatically load the
+  # +course_user+ resource.
+  def set_course_user_id_param
+    params[:course_user_id] = params[:user_id]
```
Comment (kxmbrian) [inline app/controllers/course/experience_points_records_controller.rb]: @lowjoel Is this allowed? haha

Comment (lowjoel) [inline app/controllers/course/experience_points_records_controller.rb]: https://github.com/CanCanCommunity/cancancan/blob/develop/lib/cancan/controller_additions.rb#L99

**Pull request merged**